ddocs/medic-api.json

```diff
@@ -0,0 +1 @@
+{"_id":"_design/medic-api","updates":{"update_message_task":"function (doc, request) {\n    var utils = require('medic-api-utils');\n    var data, msg;\n    var success = {\n      json: {\n        ok: true\n      }\n    };\n    var fail = function(msg, code) {\n      return [null, {\n        code: code || 500,\n        headers: {\n          \"Content-Type\" : \"application/json\"\n        },\n        body: JSON.stringify({\n          ok: false,\n          error: msg\n        })\n      }];\n    };\n    if (!doc) {\n        return fail('Document not found: ' + doc, 404);\n    }\n    try {\n        data = JSON.parse(request.body);\n    } catch(e) {\n        return fail('Parsing JSON body failed');\n    }\n    if (!data.state) {\n        return fail('Value required for state.');\n    }\n    if (!data.message_id) {\n        return fail('Message id required');\n    }\n    msg = utils.getTask(data.message_id, doc);\n    if (!data.message_id || !msg) {\n        return fail('Message not found: ' + data.message_id);\n    }\n    utils.setTaskState(msg, data.state, data.details);\n    return [\n        doc,\n        success\n    ];\n}"},"medic-api-utils":"exports.isUserAdmin = function(userCtx) {\n  return exports.isUserNationalAdmin(userCtx) || exports.isDbAdmin(userCtx);\n};\n\nexports.isUserNationalAdmin = function(userCtx) {\n  return exports.hasRole(userCtx, 'national_admin');\n};\n\nexports.isUserDistrictAdmin = function(userCtx) {\n  return exports.hasRole(userCtx, 'district_admin');\n};\n\nexports.isDbAdmin = function(userCtx) {\n  return exports.hasRole(userCtx, '_admin');\n};\n\nexports.hasRole = function(userCtx, role) {\n  var roles = (userCtx && userCtx.roles) || [];\n  return roles.indexOf(role) >= 0;\n};\n\n/**\n * Update task/message object in-place.  Used by message update functions when\n * a message's state changes. Also adds new values to state history.\n *\n * @param {Object} task\n * @param {String} state\n * @param {Any} details (optional)\n * @api public\n */\nexports.setTaskState = function(task, state, details) {\n  task.state = state;\n  task.state_details = details;\n  task.state_history = task.state_history || [];\n  task.state_history.push({\n    state: state,\n    state_details: details,\n    timestamp: new Date().toISOString()\n  });\n};\n\n/*\n * Return task object that matches message uuid or a falsey value if match\n * fails.\n *\n * @api public\n */\nexports.getTask = function(uuid, doc) {\n  for (var i in doc.tasks) {\n    for (var j in doc.tasks[i].messages) {\n      if (uuid === doc.tasks[i].messages[j].uuid) {\n        return doc.tasks[i];\n      }\n    }\n  };\n  for (var i in doc.scheduled_tasks) {\n    for (var j in doc.scheduled_tasks[i].messages) {\n      if (uuid === doc.scheduled_tasks[i].messages[j].uuid) {\n        return doc.scheduled_tasks[i];\n      }\n    }\n  };\n};\n"}
```

Comment:
This particular asset should be provided as part of the distribution for easy installation IMO.  i.e. when medic-api starts it can just expect is there and use it during bootstrap. (todo)